### PR TITLE
Update the license to try to make it detected by github.

### PR DESCRIPTION
**Pull request by tcojean**
We have a problem in Ginkgo with our license as it is not the standard text and format of a BSD 3-clause and github is not able to detect it. Here is what it looks like when it does:
https://github.com/ariya/phantomjs "View License" tab is replaced with the actual license name
https://github.com/ariya/phantomjs/blob/master/LICENSE.BSD (fancy license summary)

This is a best effort at making our license compatible with Github. I am still unsure whether Github itself will detect it. They [state in their documentation](https://help.github.com/en/articles/licensing-a-repository#detecting-a-license) that they use the tool `licensee` to detect the license. With the new version of the license, licensee detects that it is BSD-3 clause with 100% success (it did not detect anything before). Hopefully integrating this PR into our main branch will make Github recognize the license for us...

```
License:        BSD-3-Clause
Matched files:  LICENSE
LICENSE:
  Content hash:  fa22c672927af9c7334874561198799cbf4bdf31
  Attribution:   Copyright (c) 2017-2019, Karlsruhe Institute of Technology
Copyright (c) 2017-2019, Universitat Jaume I
Copyright (c) 2017-2019, University of Tennessee
  Confidence:    100.00%
  Matcher:       Licensee::Matchers::Exact
  License:       BSD-3-Clause
```

### CODE REVIEW

**Comment by tcojean** [comment]
Actually, maybe now is also a good time to discuss something related to licensing, what about external contributors? In the License, we only mention the three universities explicitly, should we also add a mention "external contributors" there? I saw that some projects do that, but how does that work then, do any person listed as contributor has copyright on the whole project, or is it only on their respective changes?

**Comment by pratikvn** [comment]
I think they retain copyright on their changes. But we should probably mention that somewhere.

**Comment by thoasm** [comment]
I think it would be nice to point out that we also have external contributors (maybe even reference the [contributors file](https://github.com/ginkgo-project/ginkgo/blob/develop/contributors.txt) in the [README.md](https://github.com/ginkgo-project/ginkgo/blob/develop/README.md)).
However, I have no idea about the laws regarding the wording of these License files, we might have to be careful what we put in there.

**Comment by tcojean** [comment]
I saw that there are multiple ways to tackle this. Some projects only seem to mention them in the readme:
https://github.com/ethereum/aleth/pull/2789/files#diff-04c6e90faac2675aa89e2176d2eec7d8

Some other projects directly include them in the License as an entity:
https://github.com/getsentry/sentry/blob/master/LICENSE
https://github.com/react-community/create-react-native-app/blob/master/LICENSE

Some others, like google, use the concept of Contributor License Agreement. It seems a bit complicated to setup, as they have a registry of cla signers somewhere and use tags `cla: yes` with bots checking for correctness.
https://opensource.google.com/docs/cla/

**Comment by thoasm** [comment]
I think adding something similar to the ` and individual contributors` from the [second mentioned license](https://github.com/getsentry/sentry/blob/master/LICENSE) would look something like this:
```
Copyright (c) 2017-2019, other contributors
```
which looks weird to me. Maybe there is a better way, but this is the first thing that came to my mind.

I think adding a section in the [README.md](https://github.com/ginkgo-project/ginkgo/blob/develop/README.md) regarding external contributors like they did in the [mentioned pull request from Etherium](https://github.com/ethereum/aleth/pull/2789/files#diff-04c6e90faac2675aa89e2176d2eec7d8) would be best suited for us, which might look like this:
```markdown
#### External Contributors
I hereby place all my contributions in this codebase under an MIT
licence, as specified [here](LICENSE).
Contributors can also add themselves in the [contributor file](contributors.txt) with
*Name Surname* (**email@domain**)
```
or similar

**Comment by tcojean** [comment]
@hartwiganzt any input on the external contributors discussion?

**Comment by hartwiganzt** [comment]
I will have a meeting with someone from licensing, maybe next week. Can it wait till then?

**Comment by thoasm** [comment]
We need to decide on that before the release, so it should be fine to postpone that for a bit (as long as we do it before the release) in my opinion.

**Comment by tcojean** [comment]
@hartwiganzt sure, that is fine

**Comment by tcojean** [comment]
This is now updated. I did the changes in the `LICENSE` and `contributors.txt` files as discussed.

**Comment by tcojean** [comment]
Here is a link with the important changes. This is basically using the file extension filter for `.md`, `.txt`, and no extensions:
https://github.com/ginkgo-project/ginkgo/pull/268/files?file-filters%5B%5D=.md&file-filters%5B%5D=.txt&file-filters%5B%5D=No+extension

File: examples/ginkgo_overhead/ginkgo_overhead.cpp [hunk 1]
```diff
@@ -72,7 +71,8 @@ env LD_LIBRARY_PATH=.:${LD_LIBRARY_PATH} ./ginkgo_overhead
 #include <iostream>
 
 
-[[noreturn]] void print_usage_and_exit(const char *name) {
+[[noreturn]] void print_usage_and_exit(const char *name)
```
Comment (hartwiganzt) [inline examples/ginkgo_overhead/ginkgo_overhead.cpp]: not sure why this change was triggered.

Comment (tcojean) [reply examples/ginkgo_overhead/ginkgo_overhead.cpp]: It's a different version of `clang-format`. I have 7.0.1-8, and I believe Fritz has an older one which indents differently. If you look at the file's history, you find this before my commit:
https://github.com/ginkgo-project/ginkgo/commit/4bec38fa133878f2fdcabd51c4009b4452a396c6#diff-b53488febfc45d801fafbd3a8b7081e4

Comment (thoasm) [reply examples/ginkgo_overhead/ginkgo_overhead.cpp]: This was probably triggered because not everyone has the same clang-format version (it was changed to the current `develop` state by @fritzgoebel since he has `clang-format` version `6.x`, and most of us have `7.0.1`).

**Comment by tcojean** [comment]
I now added a section on how to contribute and licensing details in the `README.md` file.

File: ABOUT-LICENSING.md [hunk 2]
```diff
@@ -15,9 +15,10 @@ licensing conditions (note that all of these are extra utilities, and it is
 possible to obtain a fully functional installation of Ginkgo without any of
 them). They include the following:
 
-When compiling Ginkgo with `-DBUILT_TEST=ON`, the build system will download and
-build [Google Test](https://github.com/google/googletest), and link the binary
-with Ginkgo's unit tests. Google Test is available under the following license:
+When compiling Ginkgo with `-DGINKGO_BUILD_TEST=ON`, the build system will
+download and build [Google Test](https://github.com/google/googletest), and link
+the binary with Ginkgo's unit tests. Google Test is available under the
+following license:
```
Comment (thoasm) [inline ABOUT-LICENSING.md]: Good catch, completely forgot that we might mention Cmake arguments in here.
